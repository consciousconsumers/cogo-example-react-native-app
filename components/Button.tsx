import React from 'react'
import { StyleProp, StyleSheet, TextStyle, ViewStyle, TouchableOpacity, Text } from 'react-native'

import tokens from '../design-tokens'

export type ButtonProps = {
  children: React.ReactChild
  onPress?: () => void
}

export default function Button({
  children,
  onPress
}: ButtonProps) {
  return (
    <TouchableOpacity onPress={onPress} style={styles.button}>
      <Text style={styles.text}>{children}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    lineHeight: tokens.font.leading.md,
    paddingVertical: tokens.space['3'],
    paddingHorizontal: tokens.space['6'],
    width: 'auto',
    borderRadius: tokens.radius['5'],
    backgroundColor: tokens.color.seaweed['21']
  },

  text: {
    color: tokens.color.white,
    fontSize: tokens.font.size.md,
    lineHeight: tokens.font.leading.md
  }
})
