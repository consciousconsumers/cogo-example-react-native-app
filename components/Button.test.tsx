import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('<Button />', () => {
  it('renders a default button correctly', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('renders a primary button correctly', () => {
    const button = renderer.create(<Button buttonStyle='primary' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('renders a secondary button correctly', () => {
    const button = renderer.create(<Button buttonStyle='secondary' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('renders a circular primary button correctly', () => {
    const button = renderer.create(<Button buttonStyle='primary' buttonShape='circle' icon='plus' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('renders a circular secondary button correctly', () => {
    const button = renderer.create(<Button buttonStyle='secondary' buttonShape='circle' icon='minus' />).toJSON();
    expect(button).toMatchSnapshot();
  });
});