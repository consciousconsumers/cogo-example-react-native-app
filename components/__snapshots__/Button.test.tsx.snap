// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders a circular primary button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#025D69",
      "borderRadius": 24,
      "height": 48,
      "justifyContent": "center",
      "lineHeight": 24,
      "opacity": 1,
      "paddingHorizontal": 0,
      "paddingVertical": 0,
      "width": 48,
    }
  }
>
  <View>
    <SvgMockDefault
      height={24}
      style={
        Array [
          Object {
            "color": "black",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 18,
            "lineHeight": 24,
          },
        ]
      }
      width={24}
    />
  </View>
</View>
`;

exports[`<Button /> renders a circular secondary button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderColor": "#025D69",
      "borderRadius": 24,
      "borderWidth": 1,
      "height": 48,
      "justifyContent": "center",
      "lineHeight": 24,
      "opacity": 1,
      "paddingHorizontal": 0,
      "paddingVertical": 0,
      "width": 48,
    }
  }
>
  <View>
    <SvgMockDefault
      height={24}
      style={
        Array [
          Object {
            "color": "black",
          },
          Object {
            "color": "#025D69",
            "fontSize": 18,
            "lineHeight": 24,
          },
        ]
      }
      width={24}
    />
  </View>
</View>
`;

exports[`<Button /> renders a default button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#025D69",
      "borderRadius": 24,
      "lineHeight": 24,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
      "width": "auto",
    }
  }
/>
`;

exports[`<Button /> renders a primary button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#025D69",
      "borderRadius": 24,
      "lineHeight": 24,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
      "width": "auto",
    }
  }
/>
`;

exports[`<Button /> renders a secondary button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#025D69",
      "borderRadius": 24,
      "borderWidth": 1,
      "lineHeight": 24,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
      "width": "auto",
    }
  }
/>
`;
